src/ReactDOM.re

```diff
@@ -648,7 +648,7 @@ type domProps = {
   [@mel.optional]
   acceptCharset: option(string),
   [@mel.optional]
-  action: option(string), /* uri */
+  action: option(FormData.t => Js.Promise.t(unit)), /* Since action is taken by "form" as string and React 19 accepts a callback we keep a 'action_' field to avoid a breaking change. */
```

Comment:
what do you mean by "keep a 'action_'" field? isn't the field still named `action` (without underscore?)

Reply:
I believe this is now a breaking change because we don't accept a string for `action` anymore

Reply:
I couldn't find a way to keep both (you can't have 2 mel.as with the same name), so it's either we don't support action like this (and users could cloneElement it) or we break with action.

Not fan of either

Reply:
as I understand `action` is only for `<form>`? maybe create `<Form />`, a separate component specifically for RSC apps, not to make API more complex for a majority of use cases?

Reply:
we thought we'd be able to support this by e.g. adding `@mel.unwrap` support in `@@deriving jsProperties` but it turns out it's unsound (some discussion in https://github.com/melange-re/melange/pull/1162).

it'd be kinda nice if ppxlib allowed to make the original type abstract, which would remove the unsoundness.